elastic/kibana
# [Cloud] ES endpoint discovery

In this PR I've added a new "Endpoints" item in the top bar Help menu (only in Cloud).

## Implementation

As the deployment details is consumed in 2 places (fleet "Integrations" and help menu), I've created a package to expose the UI component.
I set the ownership of the package to the team that owns the "cloud_integrations" plugin (the `core` team).
Let me know if this should be changed.

## How to test

* Add the following in `kibana.yml`

```
xpack.cloud.id: "ftr_fake_cloud_id:aGVsbG8uY29tOjQ0MyRFUzEyM2FiYyRrYm4xMjNhYmM="
```

* Modify the `maybe_add_cloud_links` file

```diff
--- a/x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/maybe_add_cloud_links.ts
+++ b/x-pack/plugins/cloud_integrations/cloud_links/public/maybe_add_cloud_links/maybe_add_cloud_links.ts
@@ -26,6 +26,7 @@ export function maybeAddCloudLinks({ core, security, cloud, share }: MaybeAddClo
   const userObservable = defer(() => security.authc.getCurrentUser()).pipe(
     // Check if user is a cloud user.
     map((user) => user.elastic_cloud_user),
+    map(() => true),
     // If user is not defined due to an unexpected error, then fail *open*.
     catchError(() => of(true)),
     filter((isElasticCloudUser) => isElasticCloudUser === true),
```

* Launch Kibana and open the "help" menu: check that the modal is correct
* Navigate to "Integrations" and check that the "Endpoints" popover is correct

## Screenshots

<img width="1445" alt="Screenshot 2023-09-25 at 12 36 44" src="https://github.com/elastic/kibana/assets/2854616/8e238c1e-1d34-41e2-bb67-6cc251c11516">


<img width="1445" alt="Screenshot 2023-09-25 at 12 37 01" src="https://github.com/elastic/kibana/assets/2854616/b4ae3765-494c-433e-af3c-b53c60ad3d8f">


<img width="1428" alt="Screenshot 2023-09-25 at 12 37 36" src="https://github.com/elastic/kibana/assets/2854616/73ab5692-39b7-44fd-be8c-d1d3958f4600">


Fixes https://github.com/elastic/kibana/issues/165019
